src/config/config_setup.cc

```diff
@@ -625,9 +625,9 @@ std::vector<std::string> ConfigArraySetup::getXmlContent(const pugi::xml_node& o
     if (result.empty()) {
         log_debug("{} assigning {} default values", xpath, defaultEntries.size());
         useDefault = true;
-        result.assign(defaultEntries.begin(), defaultEntries.end());
+        return defaultEntries;
     }
-    if (notEmpty && result.empty()) {
+    if (notEmpty) {
```

Comment:
This changes the logic if defaultEntries are not set!

Reply:
Hmm I just assumed defaultEntries was some static vector or something. I'll revert.